portfolio/src/main/webapp/pagination.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+document.addEventListener('DOMContentLoaded', () => {
+  const commentButton = new CommentButton();
+  commentButton.requestAndDisplayNewCommentsPage();
+});
+
+class CommentButton {
+  constructor() {
+    this.currentCommentsPageIndex = 0;
+    this.totalNumberOfPagesRequested = 1;
+    this.lastPage = -1;
+    this.cursorString = null;
+     
+    document.getElementById('prev-comments-button').classList.add("hidden");
+    document.getElementById('end-of-comments-sign').classList.add("hidden");
+   
+    document.getElementById('maxNum').addEventListener('input', () => {
+      document.getElementById('comments').innerHTML = "";
+      document.getElementById('prev-comments-button').classList.add("hidden");
+      if (document.getElementById('next-comments-button').classList.contains("hidden")) {
+        document.getElementById('next-comments-button').classList.remove("hidden");
+      }
+      document.getElementById('end-of-comments-sign').classList.add("hidden");
+      this.currentCommentsPageIndex = 0;
+      this.totalNumberOfPagesRequested = 1;
+      this.lastPage = -1;
+      this.requestAndDisplayNewCommentsPage();
+
+    });
+
+    document.getElementById('prev-comments-button').addEventListener('click', () => {
+      this.displayPrevPage();
+    });
+
+    document.getElementById('next-comments-button').addEventListener('click', () => {
+      this.displayNextPage();
+    });
+  }
+
+  displayNextPage() {
+    if (document.getElementById('prev-comments-button').classList.contains("hidden")) {
+      document.getElementById('prev-comments-button').classList.remove("hidden");
+    }
+
+    let commentList = document.getElementById('comments').children;
+    if (this.currentCommentsPageIndex + 1 === this.lastPage) {
+        document.getElementById('next-comments-button').classList.add("hidden");
+        document.getElementById('end-of-comments-sign').classList.remove("hidden");
+    }
+    if (this.currentCommentsPageIndex + 1 < this.totalNumberOfPagesRequested) {
+      commentList[this.currentCommentsPageIndex+1].classList.remove("hidden");
```

Comment:
i would make helper functions

hideCommentAtIndex(index) {
   const commentList = document.getElementById('comments').children;
   commentList[index].add('hidden);
}

showCommentAtIndex(index) {
   const commentList = document.getElementById('comments').children;
   commentList[index].remove('hidden);
}

that will make your code easier to read.